Adding documentation for the usage of `llama.cpp` for Qiskit Code Assistant GGUF models (#3551)
Qiskit/documentation: docs/guides/qiskit-code-assistant-local.mdx

raw_pred = model(input, **generation_kwargs)["choices"][0]["text"]
```

> ### Using the `llama.cpp` library
> 
> Another alternative is use `llama.cpp`, an open-source library that enables users to perform LLM inference on CPU with minimal setup.
> It gives you low-level control over the model execution and tipically it's run from the command line, pointing it to a local GGUF model file.

Reviewer: ```suggestion
### Use the `llama.cpp` library

Another alternative is to use `llama.cpp`, an open-source library for performing LLM inference on a CPU with minimal setup.
It provides low-level control over the model execution and is typically run from the command line, pointing to a local GGUF model file.
```

There are several ways you can use to install `llama.cpp` in your machine:

- Install llama.cpp using [brew, nix or winget](https://github.com/ggml-org/llama.cpp/blob/master/docs/install.md)
- Run with Docker: Check out the [Docker documentation by `llama.cpp` team.](https://github.com/ggml-org/llama.cpp/blob/master/docs/docker.md)
- Download pre-built binaries from the [releases page](https://github.com/ggml-org/llama.cpp/releases)
- Build from source by [cloning this repository](https://github.com/ggml-org/llama.cpp/blob/master/docs/build.md)

> Once installed you can use `llama.cpp` to interact with GGUF models in conversation mode in the following way:

Reviewer: ```suggestion
Once installed, you can use `llama.cpp` to interact with GGUF models in conversation mode as follows:
```

```bash
# Use a local model file
llama-cli -m my_model.gguf -cnv

# Or download and run a model directly from Hugging Face
llama-cli -hf Qiskit/granite-3.3-8b-qiskit-GGUF -cnv
```

You can also launch an OpenAI-compatible API server for the model in the following way:
```bash
llama-server -hf Qiskit/granite-3.3-8b-qiskit-GGUF
```

#### Advanced parameters

> The `llama-cli` program also gives you the ability to control the model generation via command-line options. For example, you can provide an initial “system” prompt using the `-p/--prompt` flag. In conversation mode (`-cnv`), this initial prompt acts as the system message. Otherwise you can simply prepend any desired instruction to your prompt text. You can also adjust sampling parameters: for instance temperature (`--temp`), top-k (`--top-k`), top-p (`--top-p`), repetition penalty (`--repeat-penalty`), the seed to use (`--seed`). Below is an example invocation using these options:

Reviewer: ```suggestion
With the `llama-cli` program, you can control the model generation using command-line options. For example, you can provide an initial “system” prompt using the `-p/--prompt` flag. In conversation mode (`-cnv`), this initial prompt acts as the system message. Otherwise, you can simply prepend any desired instruction to your prompt text. You can also adjust sampling parameters - for instance: temperature (`--temp`), top-k (`--top-k`), top-p (`--top-p`), repetition penalty (`--repeat-penalty`), and the seed to use (`--seed`). The following is an example invocation using these options:
```
